Client/mods/deathmatch/logic/CFileReader.cpp

```diff
@@ -0,0 +1,142 @@
+#include <StdInc.h>
+
+CFileReader::CFileReader(void)
+{
+    pFilePath = nullptr;
+    pFilePointer = nullptr;
+    pFileDataBuffer = nullptr;
+    u32FileLength = 0;
+    u32BytesReadFromBuffer = 0;
+}
+
+CFileReader::~CFileReader(void)
+{
+    UnloadFile();
+}
+
+BOOL CFileReader::CreateLoader(const char* pFileToRead)
+{
+    SetLoaderFilePath(pFileToRead);
+    if (!OpenFile() || !GetFileLength())
+        return FALSE;
+
+    AllocateBufferMemory();
+    pFileDataBuffer[u32FileLength] = '\0';
+    return TRUE;
+}
+
+BOOL CFileReader::LoadFile(void)
+{
+    if (!LoadToMemory())
+        return FALSE;
+
+    // std::cout << "Length: " << u32FileLength << std::endl;
```

Comment:
;)

Reply:
I'm rewriting CFileReader, thanks for the reviews. I'll let you know soon.